test/foundry/clearingHouse/ClearingHouse.priceBand.t.sol

```diff
@@ -0,0 +1,580 @@
+        clearingHouse.openPosition(
+            IClearingHouse.OpenPositionParams({
+                baseToken: address(baseToken),
+                isBaseToQuote: false,
+                isExactInput: true,
+                amount: 100 ether,
+                oppositeAmountBound: 0,
+                deadline: block.timestamp,
+                sqrtPriceLimitX96: 0,
+                referralCode: ""
+            })
+        );
+    }
+
+    // before in range, after out of range
+    // example: before: 9%, after 11% (open long position)
+    function test_revert_open_long_position_when_before_in_range_but_after_our_of_range() public {
+        // open long position => market price > index price
+        // before:
+        //  - market price: 99.9999999998
+        //  - spread: 0
+        // after:
+        //  - market price: 110.1450249998
+        //  - spread: 101450 (10.1450 %)
+        vm.prank(address(trader));
+        vm.expectRevert(bytes("EX_OPB"));
+        clearingHouse.openPosition(
+            IClearingHouse.OpenPositionParams({
+                baseToken: address(baseToken),
+                isBaseToQuote: false,
+                isExactInput: true,
+                amount: 1000 ether,
+                oppositeAmountBound: 0,
+                deadline: block.timestamp,
+                sqrtPriceLimitX96: 0,
+                referralCode: ""
+            })
+        );
+    }
+
+    // before in range, after out of range
+    // example: before: 5%, after -12% (open reverse position)
+    function test_revert_reverse_long_position_when_before_in_range_but_after_our_of_range() public {
+        // open long position => market price > index price
+        // before:
+        //  - market price: 99.9999999998
+        //  - spread: 0
+        // after:
+        //  - market price: 100.9924502498
+        //  - spread: 9924 (0.9924 %)
+        vm.prank(address(trader));
+        clearingHouse.openPosition(
+            IClearingHouse.OpenPositionParams({
+                baseToken: address(baseToken),
+                isBaseToQuote: false,
+                isExactInput: true,
+                amount: 100 ether,
+                oppositeAmountBound: 0,
+                deadline: block.timestamp,
+                sqrtPriceLimitX96: 0,
+                referralCode: ""
+            })
+        );
+
+        // open a reverse position => market price will lees than index price and price spread will out of range
+        vm.expectRevert(bytes("EX_OPB"));
+        vm.prank(address(trader));
+        clearingHouse.openPosition(
+            IClearingHouse.OpenPositionParams({
+                baseToken: address(baseToken),
+                isBaseToQuote: true,
+                isExactInput: false,
+                amount: 1500 ether,
+                oppositeAmountBound: 0,
+                deadline: block.timestamp,
+                sqrtPriceLimitX96: 0,
+                referralCode: ""
+            })
+        );
+    }
```

Comment:
why don't we direclty create a large short position to micmic "before is 5% and after is -12%"?